Update README with the latest information and doc links (#1042)
celo-org/celo-blockchain: README.md

## Celo Blockchain

Official golang implementation of the Celo blockchain, based off of the [official golang implementation of the Ethereum protocol](https://github.com/ethereum/go-ethereum).

> [![Discord](https://img.shields.io/badge/discord-join%20chat-blue.svg)](https://discord.gg/6yWMkgM)

Reviewer: ```suggestion
[![Discord](https://img.shields.io/badge/discord-join%20chat-blue.svg)](https://chat.celo.org)
```

Prebuilt [Docker](https://en.wikipedia.org/wiki/Docker_\(software\)) images are available for immediate use: [us.gcr.io/celo-testnet/celo-node](https://us.gcr.io/celo-testnet/celo-node). See [docs.celo.org/getting-started](https://docs.celo.org/getting-started) for a guide to the Celo networks and how to get started.

Documentation for Celo more generally can be found at [docs.celo.org](https://docs.celo.org/)

> Most functionality of this client is simmilar to `geth`, from which it was forked, if you do not find your question answered by Celo-specific documentation, try searching the [geth wiki](https://github.com/ethereum/go-ethereum/wiki).

Reviewer: ```suggestion
Most functionality of this client is similar to `go-ethereum`, also known as `geth`, from which it was forked. If you do not find your question answered by Celo-specific documentation, try searching the [geth wiki](https://github.com/ethereum/go-ethereum/wiki).
```

## Building the source

> Building `geth` requires both a Go (version 1.13 or later) and a C compiler and the latest Rust toolchain.

Reviewer: There's an issue which causes a test to fail on 1.14 until we merge the trivial upstream fix for it.
```suggestion
Building `geth` requires both a Go (version 1.13) and a C compiler and the latest Rust toolchain.
```

Reviewer: I thought Kobi removed the need for the Rust toolchain? not sure on progress of this..

Author: I just tested this by running a clean build on `master` and indeed the Rust compilation step no longer runs!

## Executables

The Celo blockchain client comes with several wrappers/executables found in the `cmd` directory.

| Command    | Description |
|:----------:|-------------|

> | **`geth`** | The main Celo Blockchain client. It is the entry point into the Celo network, capable of running as a full node (default), archive node (retaining all historical state), light node (retrieving data live), or lightest node. It can be used by other processes as a gateway into the Celo network via JSON RPC endpoints exposed on top of HTTP, WebSocket and/or IPC transports. `geth --help` and the [Ethereum CLI Wiki page](https://github.com/ethereum/go-ethereum/wiki/Command-Line-Options) for command line options. |

Reviewer 2: ```suggestion
| **`geth`** | The main Celo Blockchain client. It is the entry point into the Celo network, capable of running as a full node (default), archive node (retaining all historical state), light node (retrieving data live), or lightest node (retrieving minimum number of block headers to verify existing validator set). It can be used by other processes as a gateway into the Celo network via JSON RPC endpoints exposed on top of HTTP, WebSocket and/or IPC transports. `geth --help` and the [Ethereum CLI Wiki page](https://github.com/ethereum/go-ethereum/wiki/Command-Line-Options) for command line options. |
```

| `gethrpctest` | Developer utility tool to support the [ethereum/rpc-test](https://github.com/ethereum/rpc-tests) test suite which validates baseline conformity to the [Ethereum JSON RPC](https://github.com/ethereum/wiki/wiki/JSON-RPC) specs. Please see the [ethereum test suite's readme](https://github.com/ethereum/rpc-tests/blob/master/README.md) for details. |
| `rlpdump` | Developer utility tool to convert binary RLP ([Recursive Length Prefix](https://github.com/ethereum/wiki/wiki/RLP)) dumps (data encoding used by the Celo protocol both network as well as consensus wise) to user friendlier hierarchical representation (e.g. `rlpdump --hex CE0183FFFFFFC4C304050583616263`). |

## Running Celo

Please see the [docs.celo.org/getting-started](https://docs.celo.org/getting-started) for instructions on how to run a node connected the Celo network using the prebuilt Docker image.

Going through all the possible command line flags is out of scope here, please consult `geth --help` for more complete information.

> We've enumerated a few common parameter combos to get you up to speed quicklyon how you can run your own Celo blockchain client instance.

Reviewer 2: ```suggestion
We've enumerated a few common parameter combos to get you up to speed quickly on how you can run your own Celo blockchain client instance.
```
